CMakeLists.txt

```diff
@@ -1,6 +1,12 @@
 message("Running CMake on libdnf...")
 cmake_minimum_required(VERSION 3.5)
 
+option(CLANG "build with clang" OFF)
+
+if(CLANG)
+    set(CMAKE_C_COMPILER "clang")
+    set(CMAKE_CXX_COMPILER "clang++")
+endif()
```

Comment:
Do we need this option? In Fedora, we already have a [`%toolchain` macro](https://src.fedoraproject.org/rpms/redhat-rpm-config/blob/rawhide/f/macros#_39-63) that can be used to switch to Clang by setting `%toolchain` to `clang` in the spec file.

Reply:
Great, thanks, that macro does it on .spec level.

However, I'd like to have a simple switch when just building by calling cmake/make directly. I guess we could go with e.g. `CC=clang CXX=clang++ cmake ...`, but you kinda have to know about that to make it work, an explicit option would be nicer. But maybe the option is a bit redundant... WDYT?

Reply:
I think we shouldn't have it in the CMakeLists at all, and just purely control that externally. It's not like setting it is difficult. If it required a whole CMake toolchain file, I would agree with having a stuff in the CMakeLists for it, but for something as simple as overriding the compiler program name, it's not worth the debt and issues it'll cause (e.g. some distributions don't ship unversioned names for `clang`/`clang++`, etc.).

Reply:
Alright, thanks. Dropping the CLANG CMake option.